doc/tutorial.md

```diff
@@ -0,0 +1,594 @@
+
+```
+vrrp_instance VI_1 {
+    state BACKUP
+    priority 80
+    ... ...
+}
+```
+
+Start `keepalived` on both Master and Backup.
+
+```bash
+./keepalived -f /etc/keepalived/keepalived.conf
+```
+
+For **test only**, add `VIP` and *routes* to DPDK interface manually on Master (do not set VIP on both master and backup), they should be added to keepalived configure file.
+
+```bash
+./dpip addr add 192.168.100.254/32 dev dpdk0
+./dpip route add 192.168.100.0/24 dev dpdk0
+```
+
+Check if parameters just set are correct:
+
+```bash
+$ ./ipvsadm  -ln
+IP Virtual Server version 0.0.0 (size=0)
+Prot LocalAddress:Port Scheduler Flags
+  -> RemoteAddress:Port           Forward Weight ActiveConn InActConn
+TCP  192.168.100.254:80 rr
+  -> 192.168.100.2:80             FullNat 100    0          0
+  -> 192.168.100.3:80             FullNat 100    0          0
+
+$ ./dpip addr show
+inet 192.168.100.254/32 scope global dpdk0
+     valid_lft forever preferred_lft forever
+inet 192.168.100.201/32 scope global dpdk0
+     valid_lft forever preferred_lft forever sa_used 0 sa_free 1032176 sa_miss 0
+inet 192.168.100.200/32 scope global dpdk0
+     valid_lft forever preferred_lft forever sa_used 0 sa_free 1032176 sa_miss 0
+
+$ ./dpip route show
+inet 192.168.100.200/32 via 0.0.0.0 src 0.0.0.0 dev dpdk0 mtu 1500 tos 0 scope host metric 0 proto auto
+inet 192.168.100.201/32 via 0.0.0.0 src 0.0.0.0 dev dpdk0 mtu 1500 tos 0 scope host metric 0 proto auto
+inet 192.168.100.0/24 via 0.0.0.0 src 0.0.0.0 dev dpdk0 mtu 1500 tos 0 scope link metric 0 proto auto
+
+$ ./ipvsadm  -G
+VIP:VPORT            TOTAL    SNAT_IP              CONFLICTS  CONNS
+192.168.100.254:80   2
+                              192.168.100.200      0          0
+                              192.168.100.201      0          0
+```
+
+Seems good, then try access the VIP from client.
+
+```bash
+client$ curl 192.168.100.254
+Your ip:port : 192.168.100.146:42394
+```
+
+> We just explain how DPVS works with keepalived, and not verify the master/backup feature provided by keepalived works. Pls refer LVS docs if needed.
+
+<a id='dr'/>
+
+# DR Mode (one-arm)
+
+Let's make a simple example for DR mode, some user may need it.
+
+![dr-one-arm](./pics/dr-one-arm.png)
+
+To use DR:
+
+* dpvs need a LAN IP first. (for one-arm, it must be different from VIP).
+* the `RS` and `DPVS` must in same sub-network (*on-link*).
+* On `RS`: `VIP` must be added to its *lo* interface.
+* On `RS`: `arp_ignore` must be set to *lo* interface.
+
+> `DPVS` need an *RS-faced* IP it self (here means "LAN-side" IP, it's not same conception as Local-IP (LIP) used by FNAT, just an normal IP address). Because `DPVS` need communicated with `RS`es. For *one-arm*, this LAN IP and VIP are on same DPDK interface. But they cannot be same, because `VIP` will also be set on `RS`es, if we do not use a separated LAN-IP, `RS`es will not reply the ARP request. Furthermore, the LAN-IP of `DPVS` must add **before** VIP.
```

Comment:
need -> needs
it self -> itself
same -> the same
an *RS-faced*  -> a *RS-faced*
an normal -> a normal